.github/workflows/tutorials.yml

```diff
@@ -0,0 +1,138 @@
+name: Tutorials
+
+on:
+  workflow_dispatch:  # Activate this workflow manually
+  pull_request:
+    paths:
+      - 'tutorials/*.*'
+  schedule:
+    - cron: '0 0 * * *'
+
+
+jobs:
+
+  tutorials:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        script-path:
+           # 2 is very long
+        - "1_Basic_QA_Pipeline \
```

Comment:
Yep this was an opinionated choice. To recap:

- Many tutorials are very quick, in the order of 1-2 minutes. As for the tests, spinning up a runner for a 1 minute job is very wasteful and clogs the CI
- A good number of tutorials is medium-lenght, between 5 and 20 mins. For those it would be ok to have dedicated runners
- I tried to group them in almost ordinal order to avoid losing tutorials around :smile: 

Open to other opinions here, but I believe 1 runner per tutorial is not a good idea.

Reply:
Have a look here for a guess of how the CI would look like if we were to spin up one runner per tutorial (except for the notebooks-script duplication): https://github.com/deepset-ai/haystack/actions/runs/2333389651

Reply:
Thanks for the explanation, I think we can avoid the problem altogether if we dig one layer deeper. 

Right now all the tutorials are executed on a PR even if I touch only one of them. Instead of using a matrix, we could inspect the diff and find out which tutorial exactly should be executed, reducing the number of workers to possibly one.

For the cron job, we could just run them sequentially, we don't care how long it takes.

Reply:
Alright! I'll implement the diff check :+1: 

For the cron job, I'm not sure that is an option. The longer a GH runner goes, the more is likely that it will get killed, and above 1 or 2h it becomes very likely that it will die. I will now give it a try with two runners (one for scripts and one for notebooks) to see 1. how long it takes and 2. if it manages at all. I'd start thinking about a plan B though :sweat_smile:

Reply:
Jobs are [killed after 6 hours](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits), if it's killed before we should investigate whether we trigger a different limit.
Worst case, if the job starts when there's no activity, we can go back to parallel execution assuming all the workers would be idle anyways.